libraries/lib-module-manager/PluginHost.cpp

```diff
@@ -0,0 +1,220 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file PluginHost.cpp
+
+  @author Vitaly Sverchinsky
+
+  Part of lib-module-manager library
+
+**********************************************************************/
+
+#include "PluginHost.h"
+
+#include <wx/app.h>
+#include <wx/log.h>
+#include <wx/module.h>
+#include <wx/process.h>
+
+#include "BasicUI.h"
+#include "PathList.h"
+#include "FileNames.h"
+#include "ModuleManager.h"
+#include "IPCClient.h"
+#include "PlatformCompatibility.h"
+#include "PluginManager.h"
+
+namespace
+{
+   //Attempts to instantiate plugin module and put plugin descriptors into result
+   void Discover(details::PluginValidationResult& result, const wxString& providerId, const wxString& pluginPath)
+   {
+      try
+      {
+         if(auto provider = ModuleManager::Get().CreateProviderInstance(providerId, wxEmptyString))
+         {
+            TranslatableString errorMessage{};
+
+            provider->DiscoverPluginsAtPath(pluginPath, errorMessage, [&result](PluginProvider *provider, ComponentInterface *ident)
+            {
+               //Workaround: use DefaultRegistrationCallback to create all descriptors for us
+               //and then put a copy into result
+               auto id = PluginManager::DefaultRegistrationCallback(provider, ident);
+               if(const auto desc = PluginManager::Get().GetPlugin(id))
+                  result.Add(PluginDescriptor { *desc });
+               return id;
+            });
+            if(!errorMessage.empty())
+               result.SetError(errorMessage.Translation());
+         }
+         else
+            result.SetError("provider not found");
+      }
+      catch(...)
+      {
+         result.SetError("unknown error");
+      }
+   }
+}
+
+PluginHost::PluginHost()
+{
+   FileNames::InitializePathList();
+
+   wxFileName configFileName{ FileNames::Configuration() };
+   auto pConfig = std::make_unique<FileConfig>(
+      AppName, wxEmptyString, configFileName.GetFullPath(),
+      wxEmptyString, wxCONFIG_USE_LOCAL_FILE);
+   pConfig->Init();
+   InitPreferences(std::move(pConfig));
```

Comment:
In AudacityApp.cpp, `PopulatePreferences` follows `InitPreferences` and that is where translation is initialized.

That function, or pieces of it, should be moved out of AudacityApp.cpp and demoted into one of the lib-s, so it can be used here too, resolving the doubts about `Translation()` I mentioned before.